plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.test.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import { AnyApiRef, errorApiRef } from '@backstage/core-plugin-api';
+import { UrlPatternDiscovery } from '@backstage/core-app-api';
+import Router from 'react-router-dom';
+
+import { rest } from 'msw';
+import {
+  setupRequestMockHandlers,
+  TestApiProvider,
+  wrapInTestApp,
+  MockFetchApi,
+} from '@backstage/test-utils';
+import { setupServer } from 'msw/node';
+import {
+  singleBuildResponseMock,
+  entityMock,
+  buildLogResponseMock,
+} from '../../mocks/mocks';
+import { buildKiteApiRef } from '../..';
+import { BuildkiteApi } from '../../api';
+import { rootRouteRef } from '../../plugin';
+import BuildkiteBuildView from './BuildKiteBuildView';
+
+const postMock = jest.fn();
+
+const errorApiMock = { post: postMock, error$: jest.fn() };
+const discoveryApi = UrlPatternDiscovery.compile('http://exampleapi.com');
+const fetchApi = new MockFetchApi();
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: jest.fn(),
+}));
+
+const apis: [AnyApiRef, Partial<unknown>][] = [
+  [errorApiRef, errorApiMock],
+  [buildKiteApiRef, new BuildkiteApi({ discoveryApi, fetchApi })],
+];
+
+describe('BuildkiteBuildView', () => {
+  const worker = setupServer();
+  setupRequestMockHandlers(worker);
+
+  beforeEach(() => jest.resetAllMocks());
+
+  it('should display a page build and job information', async () => {
```

Comment:
nit: found a bit difficult to understand this test. Is it possible to assign the steps to variables, idk. just an idea to improve readability.

Reply:
Sometimes with these you can break it up to separate `it`s. But easier for you to know if that's very hard to do.